static/app/views/issueList/groupSearchViewTabs/draggableTabBar.tsx

```diff
@@ -166,10 +170,9 @@ export function DraggableTabBar({
   };
 
   return (
-    <Tabs>
+    <Tabs value={selectedTabKey} onChange={setSelectedTabKey}>
       <DraggableTabList
         onReorder={onReorder}
-        onSelectionChange={setSelectedTabKey}
         selectedKey={selectedTabKey}
         showTempTab={showTempTab}
         onAddView={e => handleOnAddView(e)}
```

Comment:
`setSelectedTabKey` was mistakenly put in the  `onSelectionChange` prop - this overrode the default behavior that controls link switching. The correct place to put this is in the onChange handler in the top level Tabs component.